lib/order.rb

```diff
@@ -0,0 +1,70 @@
+require_relative 'menu'
+require 'twilio-ruby'
+
+class Order
+
+  attr_reader :basket, :menu
+
+  def initialize(menu = Menu.new)
+    @menu = menu
+    @basket = []
+    @total = 0
+    @confirmed = false
+    time = (Time.now+(60*60))
+    @confirmation_text = "Thank you! Your order was placed and will be delivered before #{time.strftime("%H")}:#{time.strftime("%M")}. You ordered:"
+  end
+
+  def add_item(item, quantity = 1)
+    check_exists(item)
+    @confirmation_text += " #{quantity} portion(s) of #{item},"
+    retrieve_price(item)
+    recalculate_total(quantity)
+    basket << {item: item, quantity: quantity}
+  end
+
+  def total
+    @total
+  end
+
+  def confirmed?
+    @confirmed
+  end
+
+  def confirm
+    confirm_and_text
+  end
+
+  private
+
+  def check_exists(item)
+    item_found = false
+    @menu.list.each do |x|
```

Comment:
Because this line violates Demeter and iterates over an object inside `@menu`, that's a strong indication that this functionality belongs in `Menu`.  Hence you might have:
```ruby
def check_exists(item_name)
  fail "Item on not on the menu" unless @menu.has_dish?(item_name)
end
```